code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,780 @@
+	var/drawing_time = 5 SECONDS
+	var/max_cooldown = 30 SECONDS
+	var/cooldown = 0
+
+/obj/effect/warped_rune
+	name = "warped rune"
+	desc = "An unstable rune born of the depths of bluespace"
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "greyspace_rune"
+	move_resist = INFINITY //here to avoid the rune being moved since it only sets it's turf once when it's drawn. doesn't include admin fuckery.
+	anchored = TRUE
+	layer = MID_TURF_LAYER
+	resistance_flags = FIRE_PROOF
+	var/activated = FALSE
+	///is only used for bluespace crystal erasing as of now
+	var/storing_time = 5 SECONDS
+	///Nearly all runes needs to know which turf they are on
+	var/turf/rune_turf
+	var/deleteme = TRUE
+
+/obj/item/slimecross/warping/examine()
+	. = ..()
+	. += "It has [warp_charge] charge left"
+
+///runes can also be deleted by bluespace crystals relatively fast as an alternative to cleaning them.
+/obj/effect/warped_rune/attackby(obj/item/used_item, mob/user)
+	. = ..()
+	if(!istype(used_item,/obj/item/stack/sheet/bluespace_crystal) && !istype(used_item,/obj/item/stack/ore/bluespace_crystal))
+		return
+
+	var/obj/item/stack/space_crystal = used_item
+	if(do_after(user, storing_time,target = src)) //the time it takes to nullify it depends on the rune too
+		to_chat(user, "<span class='notice'>You nullify the effects of the rune with the bluespace crystal!</span>")
+		qdel(src)
+		space_crystal.use(1)
+		playsound(src, 'sound/effects/phasein.ogg', 20, TRUE)
+
+/obj/effect/warped_rune/acid_act()
+	. = ..()
+	visible_message("<span class='warning'>[src] has been dissolved by the acid</span>")
+	playsound(src, 'sound/items/welder.ogg', 150, TRUE)
+	qdel(src)
+
+
+///nearly all runes use their turf in some way so we set rune_turf to their turf automatically, the rune also start on cooldown if it uses one.
+/obj/effect/warped_rune/Initialize()
+	. = ..()
+	rune_turf = get_turf(src)
+	RegisterSignal(rune_turf, COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_rune)
+
+/obj/effect/warped_rune/proc/clean_rune()
+	qdel(src)
+
+///using the extract on the floor will "draw" the rune.
+/obj/item/slimecross/warping/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+
+	if(isturf(target) && locate(/obj/effect/warped_rune) in target) //check if the target is a floor and if there's a rune on said floor
+		to_chat(user, "<span class='warning'>There is already a bluespace rune here!</span>")
+		return
+
+	if(!istype(target,/turf/open/floor) && !istype(target, runepath))
+		to_chat(user, "<span class='warning'>you cannot draw a rune here!</span>")
+		return
+
+	if(istype(target, runepath)) //checks if the target is a rune and then if you can store it
+		if(do_after(user, storing_time,target = target) && warp_charge)
+			warping_crossbreed_absorb(target, user)
+			return
+
+	if(warp_charge < 1) //check if we have at least 1 charge left.
+		to_chat(user, "<span class='warning'>[src] is empty!</span>")
+		return
+
+	if(!check_cd(user))
+		return
```

Comment:
```suggestion
	if(!proximity || !check_cd(user))
		return

	if(isturf(target) && locate(/obj/effect/warped_rune) in target) //check if the target is a floor and if there's a rune on said floor
		to_chat(user, "<span class='warning'>There is already a bluespace rune here!</span>")
		return

	if(!istype(target,/turf/open/floor) && !istype(target, runepath))
		to_chat(user, "<span class='warning'>you cannot draw a rune here!</span>")
		return

	if(istype(target, runepath)) //checks if the target is a rune and then if you can store it
		if(do_after(user, storing_time,target = target) && warp_charge)
			warping_crossbreed_absorb(target, user)
			return

	if(warp_charge < 1) //check if we have at least 1 charge left.
		to_chat(user, "<span class='warning'>[src] is empty!</span>")
		return

```

Reply:
git diffs are dumb but this just combines the two empty return checks

Reply:
This suggestion will change the behavior of the runes, so I will not follow it.

Reply:
It shouldn't change it too much, it just means you won't get the other warning messages while it's on cooldown. 
Once cooldown has finished they'll appear as normal though :p

Reply:
the cd is where it needs to be in order to allow certain actions.l